nni/experiment/config/utils/internal.py

```diff
@@ -198,3 +201,37 @@ def get_ipv4_address() -> str:
     addr = s.getsockname()[0]
     s.close()
     return addr
+
+def load_experiment_config(config_json) -> typing.Union[ExperimentConfig, RetiariiExeConfig]:
+    # avoid circular import
+    from nni.nas.experiment.config import RetiariiExeConfig
+    from ..experiment_config import ExperimentConfig
+    if 'experimentType' in config_json:
+        if config_json['experimentType'] == 'hpo':
+            return ExperimentConfig(**config_json)
+        elif config_json['experimentType'] == 'nas':
+            return RetiariiExeConfig(**config_json)
+        else:
+            raise KeyError(f'Unknown experiment_type: {config_json["experimentType"]}')
```

Comment:
`KeyError` is used when key is wrong.
It should raise `ValueError` here.

Reply:
fixed